### PR TITLE
Fix: prevent StaggeredLayout floating point errors

### LINKED ISSUE #571: StaggeredLayout ItemsStretch Fill unstable when resizing
### Describe the bug

I notice this a lot more with WinUI 3, but also occurs with UWP. If you are resizing the window with the StaggeredLayout, it'll jump between where to put the items for filling the columns or not, it's probably on an edge case where it's not sure which one to do?

See the screenshot below for when it's in a bad state, but if you resize you can get to a point where the layout will flicker as you adjust the window. Recorded a video too.

FYI @Poker-sang, if you can help. I'll be pushing up an update soon with an updated CsWinRT version which should allow running this sample on main.

Otherwise just update WinAppSDK.props with the following for now locally:

```
    <PackageReference Condition="'$(HasWinUI)' == 'true'" Include="Microsoft.WindowsAppSDK" Version="1.6.241114003" />
    <PackageReference Condition="'$(HasWinUI)' == 'true'" Include="Microsoft.Windows.CsWinRt" Version="2.2.0" PrivateAssets="all"/>
```

### Steps to reproduce

```text
1. Open the primitives component
2. Run the sample app in UWP or WASDK (easier to see with WASDK)
3. Go to the StaggeredLayout sample
4. Set ItemsStretch to Fill
5. Resize window, see issue
```

### Expected behavior

No flicker/stable layout

### Screenshots

Image

https://github.com/user-attachments/assets/1e6bdfe1-5cbc-4cdd-8e30-24342e2080ef

### Code Platform

- [x] UWP
- [x] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 10 21H2 (Build 19044)
- [ ] Windows 10 22H2 (Build 19045)
- [ ] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### Other Windows Build number

_No response_

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Windows 10, version 2104 (Build 20348)
- [ ] Windows 11, version 22H2 (Build 22000)
- [ ] Other (specify)

### Other SDK version

_No response_

### Visual Studio Version

_No response_

### Visual Studio Build Number

_No response_

### Device form factor

Desktop

### Additional context

Running from main

### Help us help you

Yes, but only if others can assist.

### PR DESCRIPTION
**Pull request by Poker-sang**
## Fixes

closes #571 

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements: 

- [x] Created a feature/dev branch in your fork (vs. submitting directly from a commit on main)
- [x] Based off latest main branch of toolkit
- [x] Tested code with current supported SDKs
- [ ] New component
  - [ ] Documentation has been added
  - [ ] Sample in sample app has been added
  - [ ] Analyzers are passing for documentation and samples
  - [ ] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/CommunityToolkit/WindowsCommunityToolkit-design-assets)
- [ ] Tests for the changes have been added (if applicable)
- [ ] Header has been added to all new source files
- [x] Contains **NO** breaking changes

### CODE REVIEW

**Comment by michael-hawker** [comment]
@Poker-sang thanks for submitting a PR! 🦙❤️ Though, looks like you accidentally submitted a change to the tooling submodule?

**Comment by Poker-sang** [comment]
> @Poker-sang thanks for submitting a PR! 🦙❤️ Though, looks like you accidentally submitted a change to the tooling submodule?

Sorry I didn't notice this before so I just committed all the changes. I didn't realize that submodule changes would be committed to the main repository as well. I'm still not familiar with git's management of submodules.

File: components/Primitives/src/StaggeredLayout/StaggeredLayoutState.cs [hunk 1]
```diff
@@ -91,14 +91,14 @@ internal void ClearColumns()
...
     /// <remarks>
-    /// If all of the items have been calculated then the actual height will be returned.
-    /// If all of the items have not been calculated then an estimated height will be calculated based on the average height of the items.
+    /// If all the items have been calculated then the actual height will be returned.
+    /// If all the items have not been calculated then an estimated height will be calculated based on the average height of the items.
     /// </remarks>
     internal double GetHeight()
     {
-        double desiredHeight = Enumerable.Max(_columnLayout.Values, c => c.Height);
+        double desiredHeight = this._columnLayout.Values.Max(c => c.Height);
```
Comment (michael-hawker) [inline components/Primitives/src/StaggeredLayout/StaggeredLayoutState.cs:99]:
```suggestion
        double desiredHeight = _columnLayout.Values.Max(c => c.Height);
```
style: we generally don't use `this.` unless we need to.

File: components/Primitives/src/StaggeredLayout/StaggeredLayoutState.cs [hunk 2]
```diff
@@ -91,14 +91,14 @@ internal void ClearColumns()
...
 
-        var itemCount = Enumerable.Sum(_columnLayout.Values, c => c.Count);
+        var itemCount = this._columnLayout.Values.Sum(c => c.Count);
```
Comment (michael-hawker) [inline components/Primitives/src/StaggeredLayout/StaggeredLayoutState.cs:101]:
```suggestion
        var itemCount = _columnLayout.Values.Sum(c => c.Count);
```
style: we generally don't use `this.` unless we need to.

**Review by michael-hawker — COMMENTED** [review]
Quick style pass comment, will checkout and test out soon. Thanks!

**Comment by Poker-sang** [comment]
@michael-hawker I hadn't meant to use this. But the .editorconfig file instructed me that I should use `this`

https://github.com/CommunityToolkit/Windows/blob/c756dd53ce159144c14cfa182edab72e672699f5/.editorconfig#L99-L102

**Comment by Arlodotexe** [comment]
> @michael-hawker I hadn't meant to use this. But the .editorconfig file instructed me that I should use `this`
> 
> https://github.com/CommunityToolkit/Windows/blob/c756dd53ce159144c14cfa182edab72e672699f5/.editorconfig#L89-L92

Adding for context https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0003-ide0009#overview

**Comment by AndrewKeepCoding** [comment]
I ran the fixed sample and confirmed that the flickering is gone.

https://github.com/user-attachments/assets/c8079b4e-d7a6-4688-99cb-89cc0cd6bc63

**Comment by AndrewKeepCoding** [comment]
I didn't record a video but I also confirmed that `ItemsStrech: None` continue to work without flickering.

**Review by michael-hawker — APPROVED** [review]
Approving and merging based on our prior code reviews and discussions, and on behalf of the great validation work done by @AndrewKeepCoding, thanks Andrew! 🦙❤️

**Comment by Poker-sang** [comment]
The `ItemsStrech=None` is fixed length, so it wouldn't have flickering. Or we wouldn't be able to see it even if it did. Though thank for your test❤.

**Comment by Poker-sang** [comment]
I thought this is transient and not caused by this pr
```
Error Message:
   Assert.AreEqual failed. Expected:<2>. Actual:<1>. Unexpected token count after adding.
  Stack Trace:
     at RichSuggestBoxExperiment.Tests.Test_RichSuggestBox.<>c__DisplayClass2_0.<<Test_RichSuggestBox_DeleteTokens>b__0>d.MoveNext() in C:\a\Windows\Windows\components\RichSuggestBox\tests\Test_RichSuggestBox.cs:line 143
--- End of stack trace from previous location ---
   at CommunityToolkit.WinUI.DispatcherQueueExtensions.<>c__DisplayClass3_0.<<EnqueueAsync>b__1>d.MoveNext() in /_/components/Extensions/src/Dispatcher/DispatcherQueueExtensions.cs:line 180
--- End of stack trace from previous location ---
   at RichSuggestBoxExperiment.Tests.Test_RichSuggestBox.Test_RichSuggestBox_DeleteTokens(String token1, String token2) in C:\a\Windows\Windows\components\RichSuggestBox\tests\Test_RichSuggestBox.cs:line 133
   at Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.ThreadOperations.ExecuteWithAbortSafety(Action action)
```

**Comment by michael-hawker** [comment]
Yup, I'll rekick, we have open issues to investigate those, they're super frustrating, but intermittent, and they only seem to happen in our CI, so they've been super hard for us to understand what's happening.

**Pull request merged**